src/python/pants/backend/helm/subsystems/helm.py

```diff
@@ -87,3 +152,27 @@ def generate_exe(self, plat: Platform) -> str:
     @memoized_method
     def remotes(self) -> HelmRemotes:
         return HelmRemotes.from_dict(self._registries)
+
+
+def _cleanup_passthrough_args(args: Iterable[str]) -> tuple[list[str], list[str]]:
+    valid_args: list[str] = []
+    removed_args: list[str] = []
+
+    skip = False
+    for arg in list(args):
```

Comment:
```suggestion
    for arg in args:
```
Probably redundant? If the intention was to be defensive about other `args` types, you could change its type to `Sequence[str]`, for example.